source/DataElement/AMDSBufferGroup.h

```diff
@@ -55,12 +55,15 @@ class AMDSBufferGroup : public QObject
 public slots:
 	/// Slot which handles a request for data. The buffer group will attempt to populate the request
 	/// based on the instructions it includes. When the data request is ready the dataRequestReady signal is emitted
-	void processClientRequest(AMDSClientRequest *clientRequest);
+	void processClientRequest(AMDSClientRequest *clientRequest, bool internalRequest = false);
 
 signals:
 	/// Signal which indicates that a request for data has been processed and is ready to be sent back to the client
 	void clientRequestProcessed(AMDSClientRequest *clientRequest);
 
+	/// Handles clientRequests that have been requested internally and don't need to be routed out of the server (flattening request for trigger/dwell)
+	void internalRequestProcessed(AMDSClientRequest *clientRequest);
```

Comment:
Yep. The original design, rightly, had all of the request going out to the tcp socket for dispatch. These requests are just inside of the server and don't need to go there or waste any extra cycles determining they have no destination.